### PR TITLE
Install publisher dependencies in GHA instead of on Azure Functions host

### PR DESCRIPTION
**Pull request by andrewbranch**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -25,18 +25,21 @@ jobs:
...
-        enable-oryx-build: true
+        package: ${{ runner.temp }}/publisher
     - name: upload artifacts
-      if: always()
+      if: failure()
       uses: actions/upload-artifact@v1
       with:
         name: temp
-        path: /home/runner/work/_temp
+        path: ${{ runner.temp }}
```
Comment (sandersn) [inline .github/workflows/deploy.yml:45]: I guess runner.temp is already defined?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -25,18 +25,21 @@ jobs:
...
         app-name: types-publisher
         slot-name: production
         publish-profile: ${{ secrets.TYPESPUBLISHERFUNCTIONSPUBLISHPROFILE }}
-        package: packages/publisher
-        scm-do-build-during-deployment: true
-        enable-oryx-build: true
+        package: ${{ runner.temp }}/publisher
     - name: upload artifacts
-      if: always()
+      if: failure()
```
Comment (sandersn) [inline .github/workflows/deploy.yml:41]: does `if: failure()` mean the old thing still runs as a backup?

**Review by sandersn — APPROVED** [review]
Looks worth a try, although I have a couple of questions.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (andrewbranch) [reply .github/workflows/deploy.yml:41]: We were only doing this for debugging purposes, and you typically don't need to debug unless something goes wrong. The upload takes much longer now that the temp dir has a copy of the app in it.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (andrewbranch) [reply .github/workflows/deploy.yml:45]: Yes. I had a relevant Actions docs link open but it’s gone forever now 😅

**Review by RyanCavanaugh — APPROVED** [review]

**Pull request merged**